src/common/lock.c

```diff
@@ -280,3 +287,58 @@ lockRelease(bool failOnNoLock)
 
     FUNCTION_LOG_RETURN(BOOL, result);
 }
+
+/**********************************************************************************************************************************/
+bool
+writeLockPercentageComplete(const String *lockPath, const String *stanza, const String *execId, const double percentageComplete)
+{
+    FUNCTION_LOG_BEGIN(logLevelDebug);
+    FUNCTION_LOG_PARAM(STRING, stanza);
+    FUNCTION_LOG_PARAM(STRING, execId);
+    FUNCTION_LOG_PARAM(DOUBLE, percentageComplete);
+    FUNCTION_LOG_END();
+
+    ASSERT(lockPath != NULL);
+    ASSERT(stanza != NULL);
+    ASSERT(execId != NULL);
+
+    bool status = false;
+
+    // This context should already exist, noop if it doesn't
+    if (lockMemContext != NULL)
+    {
+        MEM_CONTEXT_BEGIN(lockMemContext)
+        {
+            // We are only interested in the backup lock file
```

Comment:
I don't think most of this is needed. We already have the backup lock file open -- so we should just be able to grab the handle and write to it. No need to check if the file exists. Instead error if the backup lock is not held.

Reply:
i.e., `CHECK(AssertError, "backup lock is not held", lockTypeHeld = lockTypeBackup)`.

Reply:
updated in 0f56ed01..1f3fbd2b